cmd/serve_test.go

```diff
@@ -125,35 +133,36 @@ func Test_serve(t *testing.T) {
 		EnableScheduler:                 true,
 		EnableMultiTenantDB:             false,
 	}
-	var err error
 	serveOpts.AnchorPlatformAPIService, err = anchorplatform.NewAnchorPlatformAPIService(httpclient.DefaultClient(), serveOpts.AnchorPlatformBasePlatformURL, serveOpts.AnchorPlatformOutgoingJWTSecret)
 	require.NoError(t, err)
 
-	crashTrackerClient, err := di.NewCrashTracker(ctx, crashtracker.CrashTrackerOptions{
+	serveOpts.CrashTrackerClient, err = di.NewCrashTracker(ctx, crashtracker.CrashTrackerOptions{
 		Environment:      serveOpts.Environment,
 		GitCommit:        serveOpts.GitCommit,
 		CrashTrackerType: "DRY_RUN",
 	})
 	require.NoError(t, err)
-	serveOpts.CrashTrackerClient = crashTrackerClient
 
 	messengerClient, err := di.NewEmailClient(di.EmailClientOptions{EmailType: message.MessengerTypeDryRun})
 	require.NoError(t, err)
 	serveOpts.EmailMessengerClient = messengerClient
 
-	smsMessengerClient, err := di.NewSMSClient(di.SMSClientOptions{SMSType: message.MessengerTypeDryRun})
+	serveOpts.SMSMessengerClient, err = di.NewSMSClient(di.SMSClientOptions{SMSType: message.MessengerTypeDryRun})
 	require.NoError(t, err)
-	serveOpts.SMSMessengerClient = smsMessengerClient
 
-	kafkaEventManager, err := events.NewKafkaProducer([]string{"kafka:9092"})
+	serveOpts.EventProducer, err = events.NewKafkaProducer([]string{"kafka:9092"})
 	require.NoError(t, err)
-	serveOpts.EventProducer = kafkaEventManager
 
 	metricOptions := monitor.MetricOptions{
 		MetricType:  monitor.MetricTypePrometheus,
 		Environment: "test",
 	}
 	mMonitorService.On("Start", metricOptions).Return(nil).Once()
+	defer mMonitorService.AssertExpectations(t)
+
+	encryptionPassphrase := keypair.MustRandom().Seed()
+	serveOpts.SignatureService = engineMocks.NewMockSignatureService(t)
+	di.SetInstance("signature_service_instance", serveOpts.SignatureService)
```

Comment:
I loved how you solved the problem of different memory addresses when asserting the `serveOpts` expectations.